cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/GceRemoteAuthenticator.java

```diff
@@ -46,22 +48,37 @@ public String getName() {
 
   @Override
   public Credential getCredentials() throws IOException {
-    return new Credential(getAccessToken().getToken(), Credential.CredentialType.EXTERNAL_BEARER);
+    return new Credential(getAccessToken(null).getToken(),
+        Credential.CredentialType.EXTERNAL_BEARER);
+  }
+
+  /**
+   * Returns the credentials for the authentication with scopes.
+   */
+  @Nullable
+  @Override
+  public Credential getCredentials(String scopes) throws IOException {
+    return new Credential(getAccessToken(scopes).getToken(),
+        Credential.CredentialType.EXTERNAL_BEARER);
   }
 
   /**
    * Returns an unexpired access token for authentication.
    */
-  private AccessToken getAccessToken() throws IOException {
+  private AccessToken getAccessToken(@Nullable String scopes) throws IOException {
     AccessToken accessToken = this.accessToken;
     if (accessToken != null && !accessToken.isExpired()) {
       return accessToken;
     }
 
-    URL url = new URL(
-        "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token");
+    String url = "http://metadata.google.internal/computeMetadata/v1/instance/"
+        + "service-accounts/default/token";
+    if (!Strings.isNullOrEmpty(scopes)) {
```

Comment:
nit: Instead of using string format, are there any native URL builders we can use?

Reply:
ack, done.